sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -177,3 +177,22 @@ case class Version() extends LeafExpression with CodegenFallback {
     UTF8String.fromString(SPARK_VERSION_SHORT + " " + SPARK_REVISION)
   }
 }
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """_FUNC_(expr) - Return readable string representation for the data type of the input.""",
+  examples = """
+      Examples:
+      > SELECT _FUNC_(1);
+       int
+      > SELECT _FUNC_(array(1));
+       array<int>
+  """,
+  since = "3.0.0")
+// scalastyle:on line.size.limit
+case class TypeOf(child: Expression) extends UnaryExpression with CodegenFallback {
```

Comment:
Can we implement codegen too? I remember there was a discussion about implementing codegen even when the expression itself barely has a benefit by doing codegen. CodegenFallback is not supported by Wholestage codegen and it can affect the general performance by disabling it.

Reply:
This is foldable so Spark should be able to turn it into literal before entering codegen.

Reply:
Oh, right! I missed it's foldable and will be constant-folded in the optimizer..

Reply:
> Oh, right! I missed it's foldable and will be constant-folded in the optimizer..

shall I rm the codegen then？

Reply:
I think it's fine to leave it.